src/chordality.jl

```diff
@@ -0,0 +1,78 @@
+"""
+    is_chordal(g)
+
+Check whether a graph is chordal.
+
+A graph is said to be *chordal* if every cycle of length `≥ 4` has a chord
+(i.e., an edge between two nodes not adjacent in the cycle).
+
+### Performance
+This algorithm is linear in the number of vertices and edges of the graph (i.e.,
+it runs in `O(nv(g) + ne(g))` time).
+
+### Implementation Notes
+`g` is chordal if and only if it admits a perfect elimination ordering—that is,
+an ordering of the vertices of `g` such that for every vertex `v`, the set of
+all neighbors of `v` that come later in the ordering forms a complete graph.
+This is precisely the condition checked by the maximum cardinality search
+algorithm [1], implemented herein.
+
+We take heavy inspiration here from the existing Python implementation in [2].
+
+Not implemented for directed graphs, graphs with self-loops, or graphs with
+parallel edges.
+
+### References
+[1] Tarjan, Robert E. and Mihalis Yannakakis. "Simple Linear-Time Algorithms to
+    Test Chordality of Graphs, Test Acyclicity of Hypergraphs, and Selectively
+    Reduce Acyclic Hypergraphs." *SIAM Journal on Computing* 13, no. 3 (1984):
+    566–79. https://doi.org/10.1137/0213035.
+[2] NetworkX Developers. "is_chordal." NetworkX 3.5 documentation. NetworkX,
+    May 29, 2025. Accessed June 2, 2025.
+    https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.chordal.is_chordal.html.
+
+# Examples
+TODO: Add examples
+"""
+function is_chordal(g::AbstractSimpleGraph)
```

Comment:
```suggestion
function is_chordal(g::AbstractGraph)
```
We can extend this to `AbstractGraph` - we don't support multigraphs anyways currently. `AbstractSimpleGraph` is more of a technical type for historical reasons. As long as you don't modify the graph we probably don't need it.

Furthermore we can use traits to check if a graph is directed or not - for an example see here: https://github.com/JuliaGraphs/Graphs.jl/blob/c001daba3ae32a8649af5e21c0c9786d9c38e082/src/biconnectivity/articulation.jl#L23

Reply:
Oh I see you are using `induced_subgraph` below - that might requite a `SimpleGraph` - perhaps there is a way to avoid that? This function is very costly as it needs to allocate a new graph.

Reply:
Thanks for the review, @simonschoelly&mdash;will respond in full later today/tomorrow. I will say for now that (I think?) `AbstractSimpleGraph` rules out the possibility of parallel edges, which would otherwise be computationally expensive to check, but I'm not sure what your thoughts on the matter are.

Reply:
I think in order to work with other graph types defined in other packages it is better to use `AbstractGraph` - for example the graphs defined here are not subtypes of `AbstractSimpleGraph`: https://github.com/JuliaGraphs/SimpleWeightedGraphs.jl

Reply:
The current `AbstractGraph` interface does not support multiple edges at all - this is something that we have to tackle in the future - but for now the assumption is that all graphs are simple graphs with potential loops.

Reply:
So you don't need to worry about that case.